pymc3/tests/test_distributions_random.py

```diff
@@ -491,43 +414,364 @@ class TestMoyal(BaseTestCases.BaseTestCase):
-        "shape, size",
-        [
-            ((2), (1)),
-            ((2), (2)),
-            ((2, 2), (2, 100)),
-            ((3, 4), (3, 4)),
-            ((3, 4), (3, 4, 100)),
-            ((3, 4), (100)),
-            ((3, 4), (1)),
-        ],
+def seeded_numpy_distribution_builder(dist_name: str) -> Callable:
+    return lambda self: functools.partial(
+        getattr(np.random.RandomState, dist_name), self.get_random_state()
+    )
+
+
+class TestGumbelDistribution(BaseTestDistribution):
+    pymc_dist = pm.Gumbel
+    pymc_dist_params = {"mu": 1.5, "beta": 3.0}
+    expected_rv_op_params = {"mu": 1.5, "beta": 3.0}
+    reference_dist_params = {"loc": 1.5, "scale": 3.0}
+    size = 15
+    reference_dist = seeded_scipy_distribution_builder("gumbel_r")
+    tests_to_run = [
+        "check_pymc_params_match_rv_op",
+        "check_rv_size",
+        "check_pymc_dist_matches_reference",
+    ]
+
+
+class TestNormalDistribution(BaseTestDistribution):
+    pymc_dist = pm.Normal
+    pymc_dist_params = {"mu": 5.0, "sigma": 10.0}
+    expected_rv_op_params = {"mu": 5.0, "sigma": 10.0}
+    reference_dist_params = {"loc": 5.0, "scale": 10.0}
+    size = 15
+    reference_dist = seeded_numpy_distribution_builder("normal")
+    tests_to_run = [
+        "check_pymc_params_match_rv_op",
+        "check_rv_size",
+        "check_pymc_dist_matches_reference",
+    ]
+
+
+class TestNormalTauDistribution(BaseTestDistribution):
+    pymc_dist = pm.Normal
+    tau, sigma = get_tau_sigma(tau=25.0)
+    pymc_dist_params = {"mu": 1.0, "sigma": sigma}
+    expected_rv_op_params = {"mu": 1.0, "sigma": 0.2}
+    tests_to_run = ["check_pymc_params_match_rv_op"]
+
+
+class TestNormalSdDistribution(BaseTestDistribution):
+    pymc_dist = pm.Normal
+    pymc_dist_params = {"mu": 1.0, "sd": 5.0}
+    expected_rv_op_params = {"mu": 1.0, "sigma": 5.0}
+    tests_to_run = ["check_pymc_params_match_rv_op"]
+
+
+class TestUniformDistribution(BaseTestDistribution):
+    pymc_dist = pm.Uniform
+    pymc_dist_params = {"lower": 0.5, "upper": 1.5}
+    expected_rv_op_params = {"lower": 0.5, "upper": 1.5}
+    tests_to_run = ["check_pymc_params_match_rv_op"]
+
+
+class TestHalfNormalDistribution(BaseTestDistribution):
+    pymc_dist = pm.HalfNormal
+    pymc_dist_params = {"sigma": 10.0}
+    expected_rv_op_params = {"mean": 0, "sigma": 10.0}
+    tests_to_run = ["check_pymc_params_match_rv_op"]
+
+
+class TestHalfNormalTauDistribution(BaseTestDistribution):
+    pymc_dist = pm.Normal
+    tau, sigma = get_tau_sigma(tau=25.0)
+    pymc_dist_params = {"sigma": sigma}
+    expected_rv_op_params = {"mu": 0.0, "sigma": 0.2}
+    tests_to_run = ["check_pymc_params_match_rv_op"]
+
+
+class TestHalfNormalSdDistribution(BaseTestDistribution):
+    pymc_dist = pm.Normal
+    pymc_dist_params = {"sd": 5.0}
+    expected_rv_op_params = {"mu": 0.0, "sigma": 5.0}
+    tests_to_run = ["check_pymc_params_match_rv_op"]
+
+
+class TestBetaAlphaBetaDistribution(BaseTestDistribution):
```

Comment:
Would add a size and matches draws test here as we are using our slightly customized method beta_clipped something

Reply:
Would also be a good example of how to test custom random generators that are not just direct calls to numpy/scipy functions

Reply:
yeah, the Discrete Weibull test should be a good example of handling custom random generators. I haven't had time to finish it yesterday. I should be able to push it today, together with the other test for the `TestBetaAlphaBetaDistribution`

Reply:
No rush, and great work ;)

Reply:
Good comments, I should have addressed them all